File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFlySpeed.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPFlySpeed implements PlayerStateProperty<Float> {
    @Override
    public String getConfigName() {
        return "fly_speed";
    }

    @Override
    public Float getDefaultSerializedProperty() {
        return 0.1F;
    }

    @Override
    public Float getSerializedPropertyFromPlayer(Player player) {
        return player.getFlySpeed();
    }

    @Override
    public void restoreProperty(Player player, Float speed) {
        player.setFlySpeed(speed);
    }

    @Override
    public Float getValueFromConfig(ConfigurationSection config) {
        return ((Double) config.getDouble(getConfigName())).floatValue();
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFlySpeed.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPFlySpeed extends PlayerStateProperty[Float]:
  override def getConfigName: String = "fly_speed"

  override def getDefaultSerializedProperty: Float = 0.1

  override def getSerializedPropertyFromPlayer(player: Player): Float =
    player.getFlySpeed

  override def restoreProperty(player: Player, speed: Float): Unit =
    player.setFlySpeed(speed)

  override def getValueFromConfig(config: ConfigurationSection): Float =
    config.getDouble(getConfigName).toFloat